draft-handrews-jri.md

```diff
@@ -0,0 +1,536 @@
+
+The value of the `"$defs"` property MUST be an object, which MUST have objects as the values of all of its properties.  JRI keywords MUST be allowed in these objects.  Aside from their use in JSON Pointers or similar structural identifiers, the property names within the `"$defs"` object MUST NOT be considered to impose specific semantics to the property values.
+
+### Referencing
+
+JRI reference usage has two components: resolving the reference target and interpreting its semantics.  JRI offers several keywords for determining a reference target.  Once determined, regardless of which keywords are used, the possible semantics are as described in {{ref-semantics}}.
+
+#### `"$ref"`
+
+The value of the `"$ref"` property MUST be a string, and MUST be a valid IRI-reference as defined by {{!RFC3987, Section 2.2}}.  The IRI produced by resolving this IRI-reference against the current base IRI identifies the reference target.
+
+By default, the reference target MAY be assumed to be of whatever media type would otherwise be present inline, and fragments MAY be evaluated accordingly.  Context specifications MAY set other requirements regarding the nature of reference targets but MUST NOT expect these requirements to be interoperable with standalone JRI implementations.
+
+#### `"$extRef"`
+
+To avoid overloading the behavior of `"$ref"` or relying on non-interoperable context-specific behavior, `"$extRef"` (for "extensible reference") allows specifying additional metadata as well as alternative mechanisms for secondary resource identification.[^90]
+
+[^90]: It is not yet clear how this should work, or how many meta-data sub-keywords should be defined in this specification.  I have asked the AsyncAPI team some clarifying questions.  An object value in which an IRI (without fragment) is provided in one member, and a (non-fragment) JSON Pointer or other fragment-substitute is provided in another would seem to solve some problems.  However, there is also a need to set a per-reference media type, and such a media type might have a fragment syntax.  We also don't want it to be easy to define both a fragment and a fragment alternative in the same reference, as the result of that would be unintuitive at best.
+
+#### Reference semantics {#ref-semantics}
+
+Reference semantics come in two varieties, object-level and keyword-level.  Object-level semantics are, under certain circumstances, both safe and interoperable in a context-independent manner.  Keyword-level semantics allow for more flexible semantic integration of referencing with a context-specification through allowing adjacent keywords.  However, this flexibility comes at the cost of interoperability, at least when adjacent keywords are actually present.
+
+##### Object-level semantics {#object-level}
+
+With object-level semantics, an object containing a JRI reference keyword is effectively replaced with its target.  This results in an evaluation path in which the reference keyword does not appear.
```

Comment:
JSON Schema uses keyword-level semantics.  As does the OAS 3.1 Reference Object (although I had not actually remembered that until someone pointed it out the other day).

Reply:
thanks -- thus I think:
> The two terms are not well defined, so it would help to define them, how they differ

Reply:
@handrews 

> JSON Schema uses keyword-level semantics. As does the OAS 3.1 Reference Object 

I wouldn't consider OAS 3.1 as keyword semantics. Keyword semantics imply that the resolved reference would replace the keyword value, but in OAS, a Reference Object merges the referred value with the additional keywords and replaces the Reference Object with the resulting value. That makes it a variation of a object semantics rather than keyword semantics.

(This is how I understood the keyword vs object semantics concept, but you defined them so please tell me if I understood wrong)

Reply:
@jdesrosiers yeah that might be more accurate.  I haven't thought about this much as I'd forgotten about the `summary` and whatever else it is until you brought it up on the setup PR for this repo.  Thanks for pointing this out.

Reply:
@handrews please define _object-level semantics_ and _keyword-level semantics_ in the **Definitions** section